frontend/src/pages/CurrentState/CurrentState.tsx

```diff
@@ -10,13 +11,38 @@ const CurrentState: React.FC = () => {
   // Accessing location/nickname information through local storage
   const location = localStorage.getItem('selectedLocation') || 'Cassie Campbell';
   const nickname = localStorage.getItem('nickname') || 'User';
+  const userID = localStorage.getItem('userID');
 
   // Defining variables for active and queue players, and unsubscribe functions
   const [activePlayers, setActivePlayers] = useState<string[]>([]);
   const [queuePlayers, setQueuePlayers] = useState<string[]>([]);
   const unsubscribeRef = useRef<(() => void) | null>(null);
   const scheduledUpdateRef = useRef<(() => void) | null>(null);  // Cache expiry timer
 
+  // Check if the user is in the queue (what if two users have the same nickname?)
+  // const joined = activePlayers.concat(queuePlayers).includes(nickname);
+
+  // Set the button to visible if the user is logged in and in the queue
+  // const [leaveButtonVisible, setLeaveButtonVisible] = useState<boolean>(!!userID && joined);
+  const [leaveButtonVisible, setLeaveButtonVisible] = useState<boolean>(!!userID);
+
+  const handleLeaveQueue = async () => {
+    try {
+      const data = await leaveQueue(location, userID);
+
+      if (data) {
+        setLeaveButtonVisible(false);
+        alert('You have successfully left the queue.');
+      } else {
+        alert('Failed to leave the queue. Please try again.');
```

Comment:
As discussed, create a page of the form ActiveView. Move the leave queue logic there.

When the user accesses this page, ensure they have a firebaseUID set in local storage with a non null value. If it is null, then redirect them to the home page where they can select location.

Otherwise, render the current state component.

Lastly, when they click on leave queue, call a commented out sign out functionality (which is a future TODO and will set the firebaseUID to null in local storage). Then, redirect the user to the home page.